src/utils/network-monitor.ts

```diff
@@ -2130,25 +2132,26 @@ export class NetworkMonitor {
    * @param interval - The interval between retries.
    */
   async retry<T>(network: string, func: () => Promise<T>, attempts = 10, interval = 5000): Promise<T | null> {
-    for (let i = 0; i < attempts; i++) {
+    let result: T | null = null
+
+    let i = 0 // declare i outside of loop so it can be used later
+    for (i; i < attempts; i++) {
       try {
-        const result = await func()
+        result = await func()
         if (result !== null) {
           return result
         }
       } catch (error: any) {
-        this.structuredLogError(network, `Attempt ${i + 1} failed: ${error.message}`)
```

Comment:
You don't want a log to keep track of how many attempts have been made anymore? I found this to be helpful before